### PR TITLE
Fix and revamp Stack dependencies

### PR DESCRIPTION
**Pull request by banacorn**
1. Fix dependency problems (mostly related to `resourcet` from `conduit` in `lsp-test`)
2. Remove extraneous stack.yaml files
3. Add and keep those stack.yaml files that should be in the CI:
    * ghc-8.4.4 - lts-12.26
    * ghc-8.6.5 - lts-14.27
    * ghc-8.8.3 - lts-16.11
    * ghc-8.10.4 - lts-17.9

I'm going to make another PR to include Stack in the CI.
So that we can solve these Stack dependency problems once and for all.

----

edit: Stack image version changes made by this PR

| HLS    | LSP (current) | This PR |
| :----- | :------------ | :------: | 
|        | 8.4.2         |  removed  | 
|        | 8.4.3         |  removed  | 
|        | 8.4.4         |  kept     | 
| 8.6.4  | 8.6.4         |  kept  |
| 8.6.5  | 8.6.5         |  kept     |
|        | 8.8.1         |  removed  | 
| 8.8.2  | 8.8.2         |  removed  | 
| 8.8.3  |              | added |
| 8.8.4  |               | added |
| 8.10.2 |               | added |
| 8.10.3 |               | added |
| 8.10.4 |               | added |

### CODE REVIEW

**Comment by jneira** [comment]
Hi, many thanks for the clean up, looks good to me, only curious about why there is a stack,yaml for 8.8,3 instead for 8.8.4

**Comment by banacorn** [comment]
I followed the versions used by the Cabal CI

https://github.com/haskell/lsp/blob/3e688d99218889cc17aaf43337ad9902aed321f5/.github/workflows/haskell.yml#L12

We can use `stack-8.8.4.yaml` instead

**Review by jneira — APPROVED** [review]
lgtm, thanks again!

**Comment by banacorn** [comment]
👀

**Comment by banacorn** [comment]
are you gonna merge it or what?

**Comment by jneira** [comment]
Sorry i lost track of this pr, thanks for pinging again. My idea was wait for another review of an actual stack user and maintainer of the repo, as we are not building with stack in ci.

**Comment by jneira** [comment]
@michaelpj are you using stack by chance?

**Comment by michaelpj** [comment]
Nope, sorry.

File: stack-8.4.4.yaml
Comment (jneira) [inline stack-8.4.4.yaml:17]: we should remove this comments?

**Comment by jneira** [comment]
I am a little bit worried for the dissymmetry between the versions supported with this pr an haskell-language-server, one of the main clients of the library. In hls we have stack-8.6.4.yaml, stack-8.8.2.yaml, stack-8.8.3.yaml, stack-8.10.2.yaml and stack-8.10.3.yaml.
[We are thinking in remove stack-8.8.2.yaml](https://github.com/haskell/haskell-language-server/issues/1863) so it could be skipped but it would be nice to keep in sync if there is no a real good reason to remove/not add it

**Comment by Ailrun** [comment]
I agree that asymmetry between this package and HLS make the situation complicated. FWIW, I think adding or removing a stack version without a really good reason is OK, but adding or removing a stack version (even with a good reason) should be handled in a synchronized manner with HLS maintainers.

**Comment by jneira** [comment]
@banacorn to be fair, the pr only removes 8.6.4 (8.8.1 and 8.8.2 too but they are not present or about to be removed in hls) so i would only ask to restore it and keep the actual ones. Other ones could be added afterwards.

File: stack-8.4.4.yaml
Comment (banacorn) [reply stack-8.4.4.yaml:17]: Yes!

**Comment by banacorn** [comment]
I'm not sure if I'm getting it right, so I made a table, please edit the **PR change** column so that I can make this PR right

| HLS    | LSP (current) | LSP (PR) | PR change     |  
| :----- | :------------ | :------: | :----------:  |
|        | 8.4.2         |  remove  |               |
|        | 8.4.3         |  remove  |               |
|        | 8.4.4         |  fix     |   (should remove comments)            |
| 8.6.4  | 8.6.4         |  remove  | should not remove!  | 
| 8.6.5  | 8.6.5         |  fix     |               |
|        | 8.8.1         |  remove  |               |
| 8.8.2  | 8.8.2         |  remove  |               |
| 8.8.3  |               |  add     |               |
| 8.8.4  |               |          |               |
| 8.10.2 |               |          |               |
| 8.10.3 |               |          |               |
| 8.10.4 |               |  add     |               |

Also, should we add 8.10.2 instead of 8.10.4? 

> I'm strongly against removing 8.10.2 support, as there is known thread-safety issue #1465 for 8.10.-8.10.4, and its fixes in 8.10.5 is not yet released...
https://github.com/haskell/haskell-language-server/issues/1863#issuecomment-846755647

**Comment by jneira** [comment]
@banacorn thanks for the nice summary, i've edited it to make clear what is our intent.
If you add the missing ones in this pr would be nice, of course.

**Comment by banacorn** [comment]
I've added those missing stack files and made a summary at the top

**Comment by jneira** [comment]
I am building in windows some of the stack.yaml's as final sanity check, will merge asap

Ideally we should have some ci jobs building the project with stack (maybe without testing). For my experience, without ci checking them, stack.yaml's get out of sync quickly.
But it is matter of another pr 😄

**Comment by jneira** [comment]
@banacorn many thanks for the pr and your patience, i've tried some of them in my local windows and all builds have been succesfull.
I hope you could continue contributing to the repo. If you have some plans to do it feel free to open or edit an issue with them.

**Pull request merged**
